docs(linter): add autogenerated usage for each rule

camchenry opened the pull request:
Generates a section on how to enable and use each rule in a config or the CLI.

<img width="746" alt="Screenshot 2025-02-08 at 10 21 51 AM" src="https://github.com/user-attachments/assets/55370f07-8335-4a1d-be94-2fb90262aa22" />
<img width="723" alt="Screenshot 2025-02-08 at 10 21 55 AM" src="https://github.com/user-attachments/assets/092acfc2-d8b6-4671-944c-94f0727ad2d2" />
<img width="719" alt="Screenshot 2025-02-08 at 10 23 23 AM" src="https://github.com/user-attachments/assets/6019f98f-a199-4369-a5ac-ee0efb3ebdf4" />
<img width="718" alt="Screenshot 2025-02-08 at 10 23 26 AM" src="https://github.com/user-attachments/assets/23c5356d-a552-4056-84e1-ea5a07b72435" />

Sysix commented:
### Merge activity

* **Feb 8, 2:47 PM EST**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

camchenry committed:
docs(linter): add autogenerated usage for each rule (#8979)

Generates a section on how to enable and use each rule in a config or the CLI.

<img width="746" alt="Screenshot 2025-02-08 at 10 21 51 AM" src="https://github.com/user-attachments/assets/55370f07-8335-4a1d-be94-2fb90262aa22" />
<img width="723" alt="Screenshot 2025-02-08 at 10 21 55 AM" src="https://github.com/user-attachments/assets/092acfc2-d8b6-4671-944c-94f0727ad2d2" />
<img width="719" alt="Screenshot 2025-02-08 at 10 23 23 AM" src="https://github.com/user-attachments/assets/6019f98f-a199-4369-a5ac-ee0efb3ebdf4" />
<img width="718" alt="Screenshot 2025-02-08 at 10 23 26 AM" src="https://github.com/user-attachments/assets/23c5356d-a552-4056-84e1-ea5a07b72435" />
